scripts/biomedical/humanMetabolomeDatabase/format_metabolite.py

```diff
@@ -0,0 +1,285 @@
+    """
+    Breaks down a list into smaller 
+    sublists, converts it into an array
+    and appends the array to the master
+    list
+    Args:
+        list_to_shard = original list
+        shard_size = size of subist
+    Returns:
+        sharded_list = master list with
+        smaller sublists 
+    """
+    sharded_list = []
+    for i in range(0, len(list_to_shard), shard_size):
+        shard = list_to_shard[i:i + shard_size]
+        arr = np.array(shard)
+        sharded_list.append(arr)
+    return sharded_list
+
+
+def add_property_matches(df, df_col_name, property_name):
+    """
+    Takes in the dataframe and property to be queried on data
+    commons and adds the dcids for the corresponding matches
+    to the original dataframe
+    Args:
+        df = name of original dataframe
+        df_col_name = column name of property to be queried
+        property_name = property to be queried
+    Returns:
+        df = modified dataframe
+    """
+    list_prop = df[[df_col_name]].T.stack().tolist()
+    if property_name == "chebiID":
+        for i in range(len(list_prop)):
+            list_prop[i] = "CHEBI:" + str(list_prop[i])
+    arr_prop = shard(list_prop, 1000)
+    for i in range(len(arr_prop)):
+        result = property_query(property_name, arr_prop[i])
+        dcid = clean_result(result)
+        df = add_query_result(df, df_col_name, dcid)
+    return df
+
+
+def main():
+    file_input = sys.argv[1]
+    file_output = sys.argv[2]
+    file_hmdb = sys.argv[3]
+
+    df = pd.read_csv(file_input, sep='\t')
+    dfh = pd.read_csv(file_hmdb)
+
+    #inchikey matches with dc
+    df = add_property_matches(df, 'inchiKey', "inChIKey")
+    #hmdb matches with dc
+    df = add_property_matches(df, 'hmdb', "humanMetabolomeDatabaseID")
+    #kegg matches with dc
+    df = add_property_matches(df, 'keggId', "keggCompoundID")
+    #chebi matches with dc
+    df = add_property_matches(df, 'cheBlId', "chebiID")
+    #drug bank matches with dc
+    df = add_property_matches(df, 'drugbank', "drugBankMetaboliteID")
+    #Add chemblID column in the dataframe and add the corresponding chembl ids
+    #for each entry
+    uni = UniChem()
+    mapping = uni.get_mapping("kegg_ligand", "chembl")
+    chembl_list = [0] * len(df['keggId'])
+    df.insert(7, 'ChEMBL', chembl_list)
+    for index, row in df.iterrows():
+        try:  #chembl_list.insert(index, mapping[row['keggId']] )
+            chembl_list[index] = mapping[row['keggId']]
+        except:
+            pass
+    df['ChEMBL'] = chembl_list
+    df['ChEMBL'] = df['ChEMBL'].replace({'0': np.nan, 0: np.nan})
+    #Get hmdb IDs from the hmdb data
+    df = name_map(df, dfh)
+    df = kegg_map(df, dfh)
+    df = chebi_map(df, dfh)
+    # Add dcids w.r.t chembl ids, hmdb and metanetx
+    for i in df.index:
+        if df.loc[i, 'ChEMBL'] != 0:
+            df.loc[i, 'Id'] = 'bio/' + str(df.loc[i, 'ChEMBL'])
+        elif ~(isNaN(df.loc[i, 'hmdb'])):
```

Comment:
are you trying to check if the number is not NaN? More commonly, we would do `elif not isNaN(df.loc[i, 'hmdb'])`